src/main.rs

```diff
@@ -280,69 +262,81 @@ fn post_admin_signin(password: Form<SingleField>, mut cookies: Cookies) -> Strin
 }
 
 #[post("/delete_report", data = "<report_id>")]
-fn post_delete_report(client: State<Mutex<Client>>, report_id: Form<SingleField>, mut cookies: Cookies) -> String {
+async fn post_delete_report(
+    db: Connection<Db>,
+    report_id: Form<SingleField>,
+    cookies: &CookieJar<'_>,
+) -> String {
     if let Some(hash) = cookies.get_private("admin_hash") {
         if hash.value() == env::var("ADMIN_HASH").unwrap() {
```

Comment:
Currently requires an env var set some other way than .env, which is inconvenient